src/input/base-input.js

```diff
@@ -128,6 +132,78 @@ class BaseInput<T: EventTarget> extends React.Component<
     this.props.onBlur(e);
   };
 
+  getInputType() {
+    // If the type prop is equal to "password" we allow the user to toggle between
+    // masked and non masked text. Internally, we toggle between type "password"
+    // and "text".
+    if (this.props.type === 'password') {
+      return this.state.isMasked ? 'password' : 'text';
+    } else {
+      return this.props.type;
+    }
+  }
+
+  renderMaskToggle() {
+    if (this.props.type !== 'password') return null;
+
+    const [MaskToggleButton, maskToggleButtonProps] = getOverrides(
+      this.props.overrides.MaskToggleButton,
+      Button,
+    );
+    const baseButtonOverrides = {
+      BaseButton: {
+        style: ({$theme}) => ({
+          color: $theme.colors.foreground,
+        }),
+      },
+    };
+    // $FlowFixMe
```

Comment:
what's up with those?

Reply:
`mergeOverrides` has some flow type deficiencies atm - pretty much anywhere we use it we have to add a fix me comment because referencing the specific override property doesn't pass flow checks.

we should look into a better solution probably.